tests/test_protocol.re

```diff
@@ -135,12 +135,17 @@ describe("protocol state", ({test, _}) => {
     expect.option(Validators.current(validators)).toBeNone();
     expect.list(Validators.to_list(validators)).toBeEmpty();
     let new_validator = Validators.{address: Address.make_pubkey()};
-    let main_op = kind =>
-      Operation.Main_chain.make(
-        ~tezos_hash=Helpers.BLAKE2B.hash(""),
-        ~tezos_index=0,
-        ~kind,
-      );
+    let main_op = {
+      let accu = ref(0);
+      kind => {
+        incr(accu);
+        Operation.Main_chain.make(
+          ~tezos_hash=Helpers.BLAKE2B.hash(""),
+          ~tezos_index=accu^,
+          ~kind,
+        );
+      };
```

Comment:
Does the accumulator need to be stored in the state somewhere? Otherwise won't it be reset to zero whenever the node restarts?

Reply:
this is a test